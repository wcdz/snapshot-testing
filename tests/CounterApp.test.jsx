import { render, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas de Snapshot en <CounterApp />', () => {
    const initialValue = 10;

    test('Debe hacer match con el snapshot', () => {
        const valorFalla = 100;
        const { container } = render(<CounterApp value={valorFalla} />);
        expect(container).toMatchSnapshot();
    });

    // Genera advertencia
    test('Debe hacer match con el snapshot con valor sin definir', () => {
        const notValue = 'No se envio/definio el value';
        const { container, getByText } = render(<CounterApp />);
        const h2Element = getByText(notValue);
        expect(h2Element.textContent).toBe(notValue);
        expect(container).toMatchSnapshot();
    });

    test('Debe hacer match con el snapshot después de hacer clic en +1', () => {
        const { container, getByText } = render(<CounterApp value={initialValue} />);
        const incrementButton = getByText('+1');
        fireEvent.click(incrementButton);
        expect(container).toMatchSnapshot();
    });

    test('Debe hacer match con el snapshot después de hacer clic en -1', () => {
        const { container, getByText } = render(<CounterApp value={initialValue} />);
        const decrementButton = getByText('-1');
        fireEvent.click(decrementButton);
        expect(container).toMatchSnapshot();
    });

    test('Debe hacer match con el snapshot después de hacer clic en Reset', () => {
        const { container, getByLabelText } = render(<CounterApp value={initialValue} />);
        const resetButton = getByLabelText('btn-reset');
        fireEvent.click(resetButton);
        expect(container).toMatchSnapshot();
    });

    test('Debe hacer match con el snapshot después de múltiples interacciones', () => {
        const { container, getByText, getByLabelText } = render(<CounterApp value={initialValue} />);
        const incrementButton = getByText('+1');
        const decrementButton = getByText('-1');
        const resetButton = getByLabelText('btn-reset');

        fireEvent.click(incrementButton);
        fireEvent.click(decrementButton);
        fireEvent.click(incrementButton);
        fireEvent.click(resetButton);

        expect(container).toMatchSnapshot();
    });

});
