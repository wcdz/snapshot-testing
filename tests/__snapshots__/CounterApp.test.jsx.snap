// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Snapshot en <CounterApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;

exports[`Pruebas de Snapshot en <CounterApp /> Debe hacer match con el snapshot con valor sin definir 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    No se envio/definio el value
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;

exports[`Pruebas de Snapshot en <CounterApp /> Debe hacer match con el snapshot después de hacer clic en +1 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    11
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;

exports[`Pruebas de Snapshot en <CounterApp /> Debe hacer match con el snapshot después de hacer clic en -1 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    9
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;

exports[`Pruebas de Snapshot en <CounterApp /> Debe hacer match con el snapshot después de hacer clic en Reset 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;

exports[`Pruebas de Snapshot en <CounterApp /> Debe hacer match con el snapshot después de múltiples interacciones 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
